Scripts/Runtime/Entities/Lifecycle/AbstractEntityLifecycleStatusSystem.cs

```diff
@@ -0,0 +1,303 @@
+using Anvil.CSharp.Collections;
+using Anvil.Unity.DOTS.Jobs;
+using System.Collections.Generic;
+using Unity.Burst;
+using Unity.Collections;
+using Unity.Entities;
+using Unity.Jobs;
+using UnityEngine;
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    /// <summary>
+    /// Provides information on the lifecycle of entities.
+    /// </summary>
+    /// <remarks>
+    /// Best practice is to extend this system and place the newly created system at a point in time when structural
+    /// changes have been made. Ex. Right after a <see cref="EntityCommandBufferSystem"/>.
+    /// Created or Imported entities will be treated as "Arrivals" to the World.
+    /// Destroyed or Evicted entities will be treated as "Departures" from the World.
+    /// Entities that are in a "CleanUp" state can be treated as "Departures" before they are actually destroyed by
+    /// Unity.
+    /// </remarks>
+    [AlwaysUpdateSystem]
+    public abstract partial class AbstractEntityLifecycleStatusSystem : AbstractAnvilSystemBase
+    {
+        private readonly List<EntityLifecycleStatus> m_EntityLifecycleStatus;
+        private readonly bool m_ShouldCleanupEntitiesCountAsDepartures;
+        private WorldEntityState m_WorldEntityState;
+        private NativeArray<JobHandle> m_Dependencies;
+        
+        /// <summary>
+        /// Creates a new <see cref="AbstractEntityLifecycleStatusSystem"/>
+        /// </summary>
+        /// <param name="shouldCleanupEntitiesCountAsDepartures">
+        /// If true, Entities that are in a CleanUp state with <see cref="ISystemStateComponentData"/>
+        /// will be considered as Departures. They will not show up later when they are actually destroyed by Unity.
+        /// </param>
+        protected AbstractEntityLifecycleStatusSystem(bool shouldCleanupEntitiesCountAsDepartures)
+        {
+            m_ShouldCleanupEntitiesCountAsDepartures = shouldCleanupEntitiesCountAsDepartures;
+            m_EntityLifecycleStatus = new List<EntityLifecycleStatus>();
+        }
+
+        protected override void OnCreate()
+        {
+            base.OnCreate();
+
+            m_WorldEntityState = WorldEntityState.GetOrCreate(World, this);
+
+            foreach (EntityLifecycleStatus entityLifecycleStatus in m_EntityLifecycleStatus)
+            {
+                entityLifecycleStatus.CreateQuery();
+            }
```

Comment:
What risk are we avoiding by creating the queries for the subclass vs just having them create and pass back the query themselves?

Reply:
I guess none. I had done it this way so that you could do the configuration and assignment of your `IEntityLifecycleStatus` objects in your constructor rather than waiting for `OnCreate`. It also reduces the work you have to do by creating the query for you.

Reply:
Ok, this does limit how advanced the query can be. It's fine for now but maybe want to consider the change in the future?

Reply:
Yeah i think it's completely fine to add in support for EntityQueryDesc in the future. Added TODO